module/proto/gravity/v1/gravity.proto

```diff
@@ -0,0 +1,92 @@
+syntax = "proto3";
+package gravity.v1;
+
+import "cosmos_proto/cosmos.proto";
+import "google/protobuf/timestamp.proto";
+import "gogoproto/gogo.proto";
+import "cosmos/base/v1beta1/coin.proto";
+import "google/protobuf/any.proto";
+
+option go_package = "github.com/cosmos/gravity-bridge/module/x/gravity/types";
+
+// EthereumEventVoteRecord is an event that is pending of confirmation by 2/3 of the signer set.
+// The event is then attested and executed in the state machine once the required
+// threshold is met.
+message EthereumEventVoteRecord {
+  google.protobuf.Any event    = 1 [
+    (cosmos_proto.accepts_interface) = "EthereumEvent"
+  ];  
+  repeated string     votes    = 2;
+  bool                accepted = 3;
+}
+
+// LatestEthereumBlockHeight defines the latest observed ethereum block height and the
+// corresponding timestamp value in nanoseconds.
+message LatestEthereumBlockHeight {
+  uint64 ethereum_height = 1;
+  uint64 cosmos_height   = 2;
+}
+
+// EthereumSigner represents a cosmos validator with its corresponding bridge operator
+// ethereum address and its staking consensus power.
+message EthereumSigner {
+  int64  power            = 1;
+  string ethereum_address = 2;
+}
+
+// UpdateSignerSetTx is the Ethereum Bridge multisig set that relays transactions
+// the two chains. The staking validators keep ethereum keys which are used to
+// check signatures on Ethereum in order to get significant gas savings.
+message UpdateSignerSetTx {
+  uint64             nonce   = 1;
+  repeated EthereumSigner signers = 2 [
+    (gogoproto.castrepeated) = "EthereumSigners", 
+    (gogoproto.nullable) = false
+  ];
+}
+
+// BatchTx represents a batch of transactions going from Cosmos to Ethereum. Batch
+// txs are are identified by a unique hash and the token contract that is shared by
+// all the SendToEthereum
+message BatchTx {
+  uint64         nonce          = 1;
+  uint64         timeout        = 2;
+  repeated bytes transactions   = 3 [
```

Comment:
@jackzampolin As I understand it, you are changing the transactions field from being an array of SendToEthereum structs, to being an array of references to SendToEthereum structs in storage. 

This will not work when it is sent over to the orchestrator, because it needs the actual structs. We could have an alternate BatchTx type which would have embedded txs, but having two types seems confusing. We could also give the orchestrator an API to get the references itself, but this seems like a lot of work for no gain.

Given the very minor storage size improvements from this array of references over an array of embedded txs, I would recommend we go back to the old design.

Reply:
This has again been changed to embed the SendToEthereumTxs thank you for pointing this out @jtremback